rudder-core/src/main/scala/com/normation/rudder/migration/XmlEntityMigration.scala

```diff
@@ -312,20 +315,31 @@ trait BatchElementMigration[T <: MigrableEntity] extends XmlFileFormatMigration
   /**
    * General algorithm: get all change request to migrate,
    * then process and save them.
-   * Return the number of change request migrated
+   * Return the number of change request migrated.
+   * The get/save is done in batch of batchSize elements
    */
-  def process() : Box[Int] = {
-    for {
-      elts     <- findAll
-      migrated <- saveResults(
-                      migrate(elts, errorLogger)
-                    , save = save
-                    , successLogger = successLogger
-                    , batchSize = batchSize
-                  )
-    } yield {
-      migrated
+  def process() : Box[MigrationProcessResult] = {
+    def recProcessOneBatch(mig: MigrationProcessResult): Box[MigrationProcessResult] = {
+      val res = for {
+        elts     <- findBatch
+        migrated <- saveResults(
+                        migrate(elts, errorLogger)
+                      , save = save
+                      , successLogger = successLogger
+                    )
+      } yield {
+        migrated
+      }
+
+      res match {
+        case Full(k) if(k < 1) => Full(mig)
+        case Full(k) => recProcessOneBatch(MigrationProcessResult(mig.totalMigrated+k, mig.nbBatches+1))
```

Comment:
I'm not sure what you meant by "as long as you are migrating, you do migrate", but yes, if there is no problem, I continue. 

Threre is already debug logs at other points of the logic, but yes, I can check and see where to add other information. You were thinking to a kind of progress log ? "I did 11 batches", "I did 12 batches", etc ?